src/libYARP_companion/src/Companion.cpp

```diff
@@ -1860,7 +1860,10 @@ int Companion::write(const char *name, int ntargets, char *targets[]) {
         std::string slash=NetworkBase::getDirectorySeparator();
         hist_file += slash;
         hist_file += "yarp_write";
-        yarp::os::mkdir(hist_file.c_str());
+        if (yarp::os::mkdir(hist_file.c_str()) != 0)
+        {
+            return 1;
```

Comment:
Most `return 1` in this file come with a ` fprintf(stderr, "error message\n");` that prints and error that explains the problem. I think it would be helpful to have it also here, printing a message with the path to the directory that it was not possible to create. A typical case in which this message would be helpful is if for some reason `getDataHome` is full or not writable.